part-2.py

```diff
@@ -4,13 +4,34 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if not len(array):
+        return False
+    elif array[0] == query:
+        return True
+    return search(array[1:], query)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    return text == text[::-1]
```

Comment:
This is definitely the easiest and probably most efficient way to solve this but I would suggest trying to solve it recursively if you can find the time.